### DISCUSSION TITLE
Maximize scratchpad on current monitor without fullscreen?

### DISCUSSION THREAD

**Discussion by Oisov** [i3/i3#4590] [Q&A] [answered]
I tried at reddit, but it seems this is the correct place =)

So I teach at a university, and the teaching is done online from my office. Here I often like to maximize a calculator to quickly show my students how to compute something.

It is there extremely important that the calculator is transparent so the maths behind it is visible, fullscreen fails to do so, In my config I have this

```bash
exec --no-startup-id alacritty --class "qalc" --config-file /home/oisov/.config/alacritty/alacritty_xxl.yml -e qalc
for_window [instance="qalc"] floating enable
for_window [instance="qalc"] move to scratchpad
for_window [instance="qalc"] resize set 100 ppt 100ppt
bindsym $mod+u exec ~/.config/i3/scratchpad_qalc.sh
```

The bash script is implemented as

```bash
#!/bin/bash
if  [ -z "$(xdotool search --classname 'qalc')" ]; then
    alacritty --class "qalc" --config-file /home/oisov/.config/alacritty/alacritty_xxl.yml -e qalc & sleep 0.5
fi

i3-msg "[instance=\"qalc\"] scratchpad show, resize set 100 ppt 100 ppt, move position 0 0"
```

Where the alacritty config file is nothing more than a much bigger fontsize.  How it looks maximized is something like this

![How it looks with a maximized calculator](https://preview.redd.it/g274mupbl6t71.png?width=2552&format=png&auto=webp&s=8da1ccc820f8e7e9817a28d7f89298ca425f5c18)

**My issue is that `move position 0 0` maximizes the calculator always on monitor 1 (of 3) instead of on 2.** _Is there a way to make the calculator appear on my current monitor, not on a specific one while still being maximized?_

Any ideas how to fix this? I use a laptop with 3 monitors docked with different screen sizes, so forcing the calculator to open on one specific monitor is not a viable solution.

**Comment by Airblader** [comment]
> fullscreen fails to do so

Why? If you're using compton/picom, you might try setting `unredir-if-possible` to `false`, perhaps that interferes.

> My issue is that move position 0 0 maximizes the calculator always on monitor 1 (of 3) instead of on 2. Is there a way to make the calculator appear on my current monitor, not on a specific one while still being maximized?

From quickly scanning the code, this may be a bug (but one that is hard to fix in a non-breaking way now); `move position` seems to always be treated as "absolute", meaning across all outputs.

That means the only way for you right now would probably be to use the IPC to figure out the position of the output and plug in those numbers into `move position`.

**Reply by Oisov** [reply]
So here it how it looks with a floating window right?

![image](https://user-images.githubusercontent.com/29121774/137478795-aff4127a-37d0-49ef-951f-1048ebda9e6d.png)

If I maximize the terminal here it how it looks

![image](https://user-images.githubusercontent.com/29121774/137479184-13dc0c68-5591-4d31-86e2-149656a68b68.png)


This is why fullscreen does not work. I am unsure about setting `unredir-if-possible` to `false`, but it seems the "issue" is that fullscreen hides everything except the maximized window. Which is a good thing normally just not in this case. I am not familiar with how the fullscreen is implemented in i3 nor how to change it for a single application.

**Comment by Oisov** [comment, marked as answer]
I was able to solve my issue using the following Python code

```bash
for_window [instance="qalc"] floating enable
for_window [instance="qalc"] move to scratchpad
for_window [instance="qalc"] resize set 100 ppt 100ppt
exec --no-startup-id alacritty --class "qalc" --config-file /home/oisov/.config/alacritty/alacritty_xxl.yml -e qalc
bindsym $mod+u exec python ~/.config/i3/qalc.py
```

Where `calc.py `is implemented as follows

```python
#!/usr/bin/env python3
import i3ipc
import subprocess
from collections import namedtuple

Pos = namedtuple("Pos", "x y")

i3 = i3ipc.Connection()


def current_output():
    workspaces = i3.get_workspaces()
    focused_workspace = [w for w in workspaces if w.focused][0]

    focused_output = i3.get_outputs()[0]
    for output in i3.get_outputs():
        if output.name == focused_workspace.output:
            return output


def get_output_pos(output):
    return Pos(output.rect.x, output.rect.y)


def calculator_running():
    return (
        subprocess.run(
            "xdotool search --classname 'qalc'",
            shell=True,
            universal_newlines=True,
        ).returncode
        == 0
    )


def run_calculator():
    subprocess.Popen(
        [
            "alacritty",
            "--class",
            "qalc",
            "--config-file",
            ALACRITTY_CONFIG,
            "-e",
            "qalc",
        ]
    )
    subprocess.run(["sleep 0.5"], shell=True)


def open_calculator(monitor):
    pos_monitor = get_output_pos(monitor)
    i3_message = f'i3-msg "[instance="qalc"] scratchpad show, resize set 100 ppt 100 ppt, move position {pos_monitor.x} {pos_monitor.y}"'
    subprocess.run([i3_message], shell=True)


ALACRITTY_CONFIG = "/home/oisov/.config/alacritty/alacritty_xxl.yml"


if __name__ == "__main__":

    print("hello")
    monitor = current_output()

    print(calculator_running())

    if not calculator_running():
        run_calculator()

    open_calculator(monitor)
```

This issue can be closed, but I feel like there should be a cleaner way than using IPC to figure out the position of the output and plug in those numbers into move position.

**Reply by Airblader** [reply]
> This issue can be closed

It's not an issue, it's a discussion – you can mark your reply as the answer, though. :-) 

> but I feel like there should be a cleaner way than using IPC to figure out the position of the output and plug in those numbers into move position.

As I said, I think this is actually a bug, because we have `move position [absolute] center` where `absolute` distinguishes current output from the screen, but `move position [absolute] x y` doesn't seem to do that if `absolute` is not given.

**Reply by Oisov** [reply]
I forgot to mention I tried using `absolute` but that gave really really strange results so I did not include it. Thanks for looking into this btw =)
